docs/backends/finagle.md

```diff
@@ -15,6 +15,8 @@ val backend = FinagleBackend()
 
 This backend depends on [finagle](https://twitter.github.io/finagle/), and offers an asynchronous backend, which wraps results in Twitter's `Future`.
 
+This backend supports host header override.
```

Comment:
As every backend supports it in some way, let's remove this - it doesn't really add any new information. The suprising behavior is when this might *not* be the case, so let's document that.